internal/webhook/core/v1/pod_webhook_test.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/opendatahub-io/odh-model-controller/internal/controller/constants"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+var _ = Describe("Pod Mutator Webhook", func() {
+	var defaulter PodMutatorDefaultor
+	var multinodePod *corev1.Pod
+	BeforeEach(func() {
+		defaulter = PodMutatorDefaultor{Client: k8sClient}
+
+		multinodePod = &corev1.Pod{
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name: "worker-container",
+						Env: []corev1.EnvVar{
+							{Name: "RAY_USE_TLS", Value: "1"},
+						},
+					},
+				},
+			},
+		}
+	})
+
+	Describe("Handle method", func() {
+		It("should add ray-tls-generator init-container to the pod if RAY_USE_TLS is set to 1", func() {
+			// mutate multinode pad
+			err := defaulter.podMutator(multinodePod)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Verify that the InitContainer was added
```

Comment:
No coverage when the pod is not a multinode one?

Reply:
updated